src/admissioncontrollerpolicy/admissionControllerPolicy.go

```diff
@@ -0,0 +1,309 @@
+		}
+		res = append(res, policyYaml)
+	}
+
+	if err := libs.UpdateOrInsertPolicyYamls(CfgDB, res); err != nil {
+		log.Error().Msgf(err.Error())
+	}
+}
+
+// GetAdmissionControllerPolicy returns admission controller policies
+func GetAdmissionControllerPolicy(namespace, clusterName, labels string) []kyvernov1.Policy {
+	var kyvernoPolicies []kyvernov1.Policy
+	filterOptions := types.PolicyFilter{
+		Cluster:   clusterName,
+		Namespace: namespace,
+		Labels:    libs.LabelMapFromString(labels),
+	}
+	policyYamls, err := libs.GetPolicyYamls(CfgDB, types.PolicyTypeAdmissionController, filterOptions)
+	if err != nil {
+		log.Error().Msgf("fetching policy yaml from DB failed err=%v", err.Error())
+		return nil
+	}
+	for _, policyYaml := range policyYamls {
+		var kyvernoPolicy kyvernov1.Policy
+		err := yaml.Unmarshal(policyYaml.Yaml, &kyvernoPolicy)
+		if err != nil {
+			log.Error().Msgf("unmarshalling policy yaml failed err=%v", err.Error())
+			continue
+		}
+		kyvernoPolicies = append(kyvernoPolicies, kyvernoPolicy)
+	}
+	return kyvernoPolicies
+}
+
+// ConvertPoliciesToWorkerResponse converts kyverno policies to worker response
+func ConvertPoliciesToWorkerResponse(policies []kyvernov1.Policy) *wpb.WorkerResponse {
+	var response wpb.WorkerResponse
+
+	for i := range policies {
+		kyvernoPolicy := wpb.Policy{}
+
+		val, err := json.Marshal(&policies[i])
+		if err != nil {
+			log.Error().Msgf("kyvernoPolicy json marshal failed err=%v", err.Error())
+			continue
+		}
+		kyvernoPolicy.Data = val
+
+		response.AdmissionControllerPolicy = append(response.AdmissionControllerPolicy, &kyvernoPolicy)
+	}
+	response.Res = "OK"
+
+	return &response
+}
+
+// AutoGenPrecondition generates a preconditions matching on particular labels for kyverno policy
+func AutoGenPrecondition(templateKey string, labels types.LabelMap, precondition apiextensions.JSON) apiextensions.JSON {
+	preconditionMap := precondition.(map[string]interface{})
+	for key, value := range labels {
+		newPrecondition := map[string]interface{}{
+			"key":      "{{ request.object.spec." + templateKey + ".metadata.labels." + key + " || '' }}",
+			"operator": "Equals",
+			"value":    value,
+		}
+		existingSlice := preconditionMap["all"].([]interface{})
+		preconditionMap["all"] = append(existingSlice, newPrecondition)
+	}
+	return preconditionMap
+}
+
+// AutoGenPattern generates a pattern changing validation pattern from Pod to high level controller
+func AutoGenPattern(templateKey string, pattern apiextensions.JSON) apiextensions.JSON {
+	newPattern := map[string]interface{}{
+		"spec": map[string]interface{}{
+			templateKey: pattern,
+		},
+	}
+	return newPattern
+}
+
+// ShouldSATokenBeAutoMounted returns true if service account token should be auto mounted
+func ShouldSATokenBeAutoMounted(namespace string, labels types.LabelMap) bool {
+	client := cluster.ConnectK8sClient()
+
+	podList, err := client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
```

Comment:
can we have this handled in the [k8sClientHandler.go](https://github.com/accuknox/discovery-engine/blob/dev/src/cluster/k8sClientHandler.go) file?

Reply:
Yes, the function can be transferred to that file, but it won't make much sense as this is something particular to admission controller policies. We just connect to the client to list pods.